### PR TITLE
Delete the output files, not whatever is in the current directory.

### PR DESCRIPTION
**Pull request by tmooney**
This implements my last comment in #628.

The test data is updated to expect the files to be removed as well :smile:

### CODE REVIEW

File: lib/perl/Genome/Model/ClinSeq/Command/IdentifyLoh.pm [hunk 1]
```diff
@@ -255,12 +255,14 @@ sub filter_loh {
...
-        " *.lc *.Germline *.removed *.formatted" .
-        " *.hc.err *.other";
-    for my $file (glob $unwanted) {
-        $self->status_message("unlinking $file");
-        unlink $file;
+    my @patterns = qw(*.Somatic* *.readcounts *.hc *.LOH
+        *.lc *.Germline *.removed *.formatted
+        *.hc.err *.other);
+    for my $pattern (@patterns) {
+        for my $file (glob File::Spec->join($self->outdir, $pattern)) {
```
Comment (acoffman) [inline lib/perl/Genome/Model/ClinSeq/Command/IdentifyLoh.pm]: Super nitpicky and maybe not worth implementing, but a `map` would eliminate the nested loops here.

````perl
my @files = map { glob File::Spec->join($self->outdir, $_) } qw(*.Somatic* *.readcounts *.hc *.LOH *.lc *.Germline *.removed *.formatted *.hc.err *.other);
$self->status_message("unlinking %s", join(',' @files));
unlink @files
````
You'd lose the ability to tell which ones failed or succeeded, but we aren't currently checking that anyways.

**Comment by acoffman** [comment]
+1 with or without my loop flattening suggestion

**Comment by nnutter** [comment]
:+1:

**Comment by acoffman** [comment]
I'm going to merge this in @tmooney's absence since he won't be back until Monday and this fixes a potentailly destructive bug.

**Pull request merged**